src/libraries/System.Net.Requests/src/System/Net/HttpWebRequest.cs

```diff
@@ -1139,6 +1145,8 @@ public Stream GetRequestStream(out TransportContext? context)
 
         public override IAsyncResult BeginGetRequestStream(AsyncCallback? callback, object? state)
         {
+            _endGetRequestStreamCalled = false;
```

Comment:
Can you move this under `_beginGetRequestStreamCalled` check? If we're already in progress of an async operation, we shouldn't touch this. Also, we should make this thread-safe, as we're already doing on line 1152.